lnwallet/size.go

```diff
@@ -337,42 +343,67 @@ func (twe *TxWeightEstimator) AddP2PKHInput() {
 }
 
 // AddP2WKHInput updates the weight estimate to account for an additional input
-// spending a P2PWKH output.
+// spending a native P2PWKH output.
 func (twe *TxWeightEstimator) AddP2WKHInput() {
 	twe.AddWitnessInput(P2WKHWitnessSize)
 }
 
 // AddWitnessInput updates the weight estimate to account for an additional
-// input spending a pay-to-witness output. This accepts the total size of the
-// witness as a parameter.
+// input spending a native pay-to-witness output. This accepts the total size
+// of the witness as a parameter.
 func (twe *TxWeightEstimator) AddWitnessInput(witnessSize int) {
 	twe.inputSize += InputSize
 	twe.inputWitnessSize += witnessSize
 	twe.inputCount++
 	twe.hasWitness = true
 }
 
+// AddNestedP2WKHInput updates the weight estimate to account for an additional
+// input spending a P2SH output with a nested P2WKH redeem script.
+func (twe *TxWeightEstimator) AddNestedP2WKHInput() {
+	twe.inputSize += InputSize + P2WPKHSize
```

Comment:
This should also add a data push op as the sig script will contain a _single_ pushing, being the witness program.

Reply:
The byte is included in `InputSize` already.